transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultMigrationManager.java

```diff
@@ -94,7 +94,21 @@ class DefaultMigrationManager {
               "Update length of invocation column on outbox for MySQL dialects only.",
               "ALTER TABLE TXNO_OUTBOX MODIFY COLUMN invocation MEDIUMTEXT",
               Map.of(
-                  Dialect.POSTGRESQL_9, "", Dialect.H2, "", Dialect.ORACLE, "SELECT * FROM dual")));
+                  Dialect.POSTGRESQL_9, "", Dialect.H2, "", Dialect.ORACLE, "SELECT * FROM dual")),
+          new Migration(
+              9,
+              "Add createTime column to outbox",
+              "ALTER TABLE TXNO_OUTBOX ADD COLUMN createTime TIMESTAMP(6) NULL AFTER invocation",
```

Comment:
`TIMESTAMP(6)` is millisecond accuracy based on the system clock of the system writing the record. If you have two boxes writing records at roughly the same time, the ordering will be easily screwed up by clock skew.

To my mind, the only way to guarantee order is to have a group-specific incrementing counter with write-locking semantics, e.g:

```
CREATE TABLE PARTITION_SEQ (
  groupId VARCHAR(250) NOT NULL PRIMARY KEY,
  nextSequence BIGINT NOT NULL
)
```
and
```
SELECT nextSequence FROM PARTITION_SEQ WHERE groupId = ? FOR UPDATE
```
then write your record then
```
UPDATE PARTITION_SEQ SET nextSequence = ?
```
This all has a big negative: there is big contention on write. But I don't think it's avoidable.

Reply:
Wouldn't a single global [sequence](https://www.postgresql.org/docs/current/sql-createsequence.html) (`CREATE SEQUENCE ordered_sequence` + `INSERT INTO ... nextval('ordered_sequence')`) to replace the `createTime` column do the same job with minimum complexity?

I'm actually not sure why using a sequence wasn't the approach taken in the original pull request (https://github.com/gruelbox/transaction-outbox/pull/252). With an 8-byte BIGINT wraparound seems unlikely to become an issue.

Reply:
@badgerwithagun I guess having a sequence per group ID could result in a lot of sequences if the following use case is supported:

Suppose you have on the order of a million entities for which you want to publish CRUD events. The events per entity need to be ordered, but the order of events for different entities do not matter. You would then use the entity ID as group ID resulting in on the order of a million sequences?

Having just one sequence as @markushc suggests would be simpler, but I don't know if it would be a critical performance limitation. What if an outbox entry is assigned a sequence number if and only if its group ID is non-null? On the other hand, I don't know if there is a use case for combining `schedule` calls with and without group ID even though the current API would allow for that.

Reply:
> If you have two boxes writing records at roughly the same time, the ordering will be easily screwed up by clock skew.

The branch has now been updated to use the database timestamp for initializing the task creation time to eliminate the possibility of clock skew between nodes. Trying to stay with the approach of ordering based on creation time to avoid the complexity related to group-specific locks.